### ISSUE TITLE
Loss Becomes 'nan' when Focal Loss is Used in RetinaNet

### ISSUE THREAD

**Issue by DrX2020** [open-mmlab/mmdetection#7490]
By default, loss_cls in RetinaNet is Focal Loss. Actually RetinaNet was born with Focal Loss, both proposed in the same paper. When training RetinaNet, I found that loss_cls, loss_bbox and loss became 'nan' sometimes. Reducing nms iou_threshold seemed to exacerbate the problem. Maintaining default nms iou_threshold might ease the problem but there is still a chance of 'nan' losses. I wonder how to solve the problem.

**Comment by ZwwWayne** [comment]
Hi @DrX2020 ,
You may first try to add pdb to check the exact place that causes the nan of loss_cls. There might be some issues that makes the training unstable. You can try to visualize the assignment results and see if there is anything that makes the model hard to learn.

**Comment by dr-GitHub-account** [comment]
Thank you!

**Comment by YiqunChen1999** [comment]
Hi, did you solve this? I met the same problem with command `CUDA_VISIBLE_DEVICES=0,1,2,3 PORT=19191 tools/dist_train.sh configs/retinanet/retinanet_r50_fpn_mstrain_640-800_3x_coco.py 4`. It looks good at the beginning, but after 1000 iterations the loss (loss_cls, loss_bbox, loss) suddenly became nan: 
```
2022-04-06 14:13:37,121 - mmdet - INFO - Epoch [1][900/43976]   lr: 1.000e-02, eta: 1 day, 5:48:10, time: 0.192, data_time: 0.009, memory: 3483, loss_cls: 1.1372, loss_bbox: 0.5745, loss: 1.7117
2022-04-06 14:13:46,571 - mmdet - INFO - Epoch [1][950/43976]   lr: 1.000e-02, eta: 1 day, 5:41:16, time: 0.189, data_time: 0.008, memory: 3483, loss_cls: 1.1337, loss_bbox: 0.5769, loss: 1.7106
2022-04-06 14:13:55,529 - mmdet - INFO - Exp name: retinanet_r50_fpn_mstrain_640-800_3x_coco.py
2022-04-06 14:13:55,529 - mmdet - INFO - Epoch [1][1000/43976]  lr: 1.000e-02, eta: 1 day, 5:30:36, time: 0.179, data_time: 0.007, memory: 3483, loss_cls: nan, loss_bbox: nan, loss: nan
2022-04-06 14:14:04,912 - mmdet - INFO - Epoch [1][1050/43976]  lr: 1.000e-02, eta: 1 day, 5:24:35, time: 0.188, data_time: 0.007, memory: 3483, loss_cls: nan, loss_bbox: nan, loss: nan
2022-04-06 14:14:14,418 - mmdet - INFO - Epoch [1][1100/43976]  lr: 1.000e-02, eta: 1 day, 5:20:04, time: 0.190, data_time: 0.007, memory: 3483, loss_cls: nan, loss_bbox: nan, loss: nan                                                                                 
```

**Comment by BIGWangYuDong** [comment]
maybe the learning rate is too large.
